lib/kube_link_generator.rb

```diff
@@ -63,10 +64,16 @@ def get_pods_for_role(role_name, job, options = {})
           pods = _get_pods_for_role(role_name, sts_image)
           good_pods = pods.select do |pod|
             next false unless pod.status.podIP
-            next true if pod.metadata.annotations["skiff-exported-properties-#{job}"]
+            begin
+              secret = client.get_secret("#{pod.metadata.name}-#{pod.metadata.uid}", namespace)
+              next true if secret.data["skiff-exported-properties-#{job}"]
 
-            # Fall back to non-job-specific properties, for upgrades from older versions
-            pod.metadata.annotations['skiff-exported-properties']
+            rescue
+              next true if pod.metadata.annotations["skiff-exported-properties-#{job}"]
+
+              # Fall back to non-job-specific properties, for upgrades from older versions
```

Comment:
I don't think we need _this_ fallback, because we don't support upgrades from anything less than one version behind.

Doesn't hurt, though.

Reply:
Agreed, will remove.